gnucash/import-export/import-main-matcher.c

```diff
@@ -1559,49 +1559,47 @@ get_required_color (const gchar *class_name)
 static void
 remove_child_row (GtkTreeModel *model, GtkTreeIter *iter)
 {
-    if (gtk_tree_model_iter_has_child (model, iter))
+    while (gtk_tree_model_iter_has_child (model, iter))
     {
         GtkTreeIter  child;
         gtk_tree_model_iter_nth_child (model, &child, iter, 0);
         gtk_tree_store_remove (GTK_TREE_STORE(model), &child);
     }
 }
 
+static void add_row (GtkTreeModel *model, GtkTreeIter *iter, const char *date,
+                     const char *account, const char *amount, const char *narrative)
+{
+    GtkTreeStore *store = GTK_TREE_STORE (model);
+    GtkTreeIter child;
+    gtk_tree_store_append (store, &child, iter);
+    gtk_tree_store_set (store, &child, DOWNLOADED_COL_DATE_TXT, date, -1);
+    gtk_tree_store_set (store, &child, DOWNLOADED_COL_ACCOUNT, account, -1);
+    gtk_tree_store_set (store, &child, DOWNLOADED_COL_AMOUNT, amount, -1);
+    gtk_tree_store_set (store, &child, DOWNLOADED_COL_DESCRIPTION, narrative, -1);
+    gtk_tree_store_set (store, &child, DOWNLOADED_COL_ENABLE, FALSE, -1);
+}
+
```

Comment:
You should combine the gtk_tree_store_set's to one but with my suggested changes this function is not required.